lib/modules/embeddings/topic.rb

```diff
@@ -33,55 +33,73 @@ def symmetric_semantic_search(model, topic)
+        begin
+          candidate_ids =
+            DiscourseAi::Database::Connection
+              .db
+              .query(<<~SQL, query_embedding: embedding, limit: limit, offset: offset)
+                SELECT
+                  topic_id
+                FROM
+                  topic_embeddings_#{model.name.underscore}
+                ORDER BY
+                  embedding #{model.pg_function} '[:query_embedding]'
+                LIMIT :limit
+                OFFSET :offset
+              SQL
+              .map(&:topic_id)
+        rescue PG::Error => e
+          raise StandardError, "No embeddings found for topic #{topic.id}" if candidate_ids.empty?
+        end
 
         candidate_ids
       end
 
       private
 
       def query_symmetric_embeddings(model, topic)
-        DiscourseAi::Database::Connection.db.query(<<~SQL, topic_id: topic.id).map(&:topic_id)
-          SELECT
-            topic_id
-          FROM
-            topic_embeddings_#{model.name.underscore}
-          ORDER BY
-            embedding #{model.pg_function} (
-              SELECT
-                embedding
-              FROM
-                topic_embeddings_#{model.name.underscore}
-              WHERE
-                topic_id = :topic_id
-              LIMIT 1
-            )
-          LIMIT 100
-        SQL
+        begin
+          DiscourseAi::Database::Connection.db.query(<<~SQL, topic_id: topic.id).map(&:topic_id)
+            SELECT
+              topic_id
+            FROM
+              topic_embeddings_#{model.name.underscore}
+            ORDER BY
+              embedding #{model.pg_function} (
+                SELECT
+                  embedding
+                FROM
+                  topic_embeddings_#{model.name.underscore}
+                WHERE
+                  topic_id = :topic_id
+                LIMIT 1
+              )
+            LIMIT 100
+          SQL
+        rescue PG::Error => e
+          Rails.logger.error(
+            "Error #{e} querying embeddings for topic #{topic.id} and model #{model.name}",
+          )
+        end
       end
 
       def persist_embedding(topic, model, embedding)
-        DiscourseAi::Database::Connection.db.exec(<<~SQL, topic_id: topic.id, embedding: embedding)
-            INSERT INTO topic_embeddings_#{model.name.underscore} (topic_id, embedding)
-            VALUES (:topic_id, '[:embedding]')
-            ON CONFLICT (topic_id)
-            DO UPDATE SET embedding = '[:embedding]'
-          SQL
+        begin
+          DiscourseAi::Database::Connection.db.exec(
+            <<~SQL,
+              INSERT INTO topic_embeddings_#{model.name.underscore} (topic_id, embedding)
+              VALUES (:topic_id, '[:embedding]')
+              ON CONFLICT (topic_id)
+              DO UPDATE SET embedding = '[:embedding]'
+            SQL
+            topic_id: topic.id,
+            embedding: embedding,
+          )
+        rescue PG::Error => e
+          Rails.logger.error(
```

Comment:
How will this recover? Do we have an automatic backfill job?

Reply:
Yes, we retry on next pageview on the same topic automatically.